src/bidi.jl

```diff
@@ -0,0 +1,31 @@
+import Base: size, getindex, setindex!
+
+#z indices are flipped for even-numbered timepoints
+struct BidiImageArray{T} <: AbstractArray{T,4}
+    A::AbstractArray{T,4}
+    z_size::Int
+end
+
+BidiImageArray(A::AbstractArray{T,4}) where {T} = BidiImageArray(A, size(A,3))
+
+size(B::BidiImageArray) = size(B.A)
+
+Base.IndexStyle(::Type{<:BidiImageArray}) = IndexCartesian()
+
+function getindex(B::BidiImageArray{T}, I::Vararg{Int, 4}) where {T}
```

Comment:
don't think you need the `T`

Reply:
I've been confused about this, because I've found some cases in which `SomeType{T} where {T}` is not equivalent to `SomeType`.  I don't understand why that is.

Reply:
AFAIK that's only when it's a type parameter. Consider the following definitions:
```julia
return1(::Type{BidiImageArray{T}}) where T = 1
return2(::Type{BidiImageArray{T} where T}) = 2
return3(::Type{BidiImageArray}) = 3
return4(::Type{<:BidiImageArray}) = 4
```
`return1` accepts `BidiImageArray{Int}` but not plain `BidiImageArray`: it requires you to have concrete `T` because the `where` is outside the curly brackets. `return2` and `return3` have the opposite behavior, and are completely equivalent to one another (I think this makes sense, but if you don't, see [this](https://docs.julialang.org/en/latest/manual/types/#UnionAll-Types-1)). `return4` accepts both `BidiImageArray{Int}` and `BidiImageArray`.

If this is unfamiliar also see https://docs.julialang.org/en/latest/manual/types/#Parametric-Composite-Types-1.

Reply:
Thanks a lot!  I had read the UnionAll types section in the docs, but it only fully sunk in just now.  I had it completely wrong, thought that `return1` and `return2` were equivalent and `return3` and `return4` were likewise equivalent.